compiler/src/iree/compiler/Dialect/Flow/Transforms/SetEncoding.cpp

```diff
@@ -0,0 +1,218 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===--------------- SetEncoding.cpp -------------------------------------===//
+// Sets the encoding for compute operations to allow execution of the
+// operations in tiled layouts.
+//===---------------------------------------------------------------------===//
+
+#include "iree-dialects/Dialect/LinalgExt/IR/LinalgExtDialect.h"
+#include "iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.h"
+#include "iree-dialects/Dialect/LinalgExt/Utils/Utils.h"
+#include "iree/compiler/Dialect/Flow/Transforms/PassDetail.h"
+#include "iree/compiler/Dialect/Flow/Transforms/Passes.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+namespace Flow {
+
+//===---------------------------------------------------------------------===//
+// Utility functions
+//===---------------------------------------------------------------------===//
+
+// Returns the element type of `t` if it is a `ShapedType`, else return
+// `t` itself.
+Type getElementTypeOrType(Type t) {
+  if (auto shapedType = t.dyn_cast<ShapedType>()) {
+    return shapedType.getElementType();
+  }
+  return t;
+}
+
+/// Returns a constant 0 of type `elementType`.
+static FailureOr<Value> getZero(OpBuilder &builder, Location loc,
+                                Type elementType) {
+  Attribute zeroVal =
+      TypeSwitch<Type, Attribute>(elementType)
+          .Case<FloatType>([&](FloatType floatType) -> Attribute {
+            return builder.getFloatAttr(floatType, 0);
+          })
+          .Case<IntegerType>([&](IntegerType intType) -> Attribute {
+            return builder.getIntegerAttr(intType, 0);
+          })
+          .Default([](Type type) { return nullptr; });
+  if (!zeroVal) return failure();
+  return builder.create<arith::ConstantOp>(loc, zeroVal, elementType)
```

Comment:
(would be cool if there was an arith::ConstantZeroOp helper for this - I feel like I've written this a half dozen times and I'm sure there's a lot more hiding around!)

Reply:
Found it. It is `builder.getZeroAttr`. Changing the code to use that.